### PR TITLE
[GFX11] Add support for gfx1101 target ID

### PR DESCRIPTION
**Pull request by junliume**

### CODE REVIEW

**Review by muralinr — APPROVED** [review]
LGTM

File: test/handle_test.cpp [hunk 1]
```diff
@@ -242,9 +242,10 @@ void test_arch_name()
                        "gfx900",
                        "gfx803",
                        "gfx1030",
+                       "gfx1031",
                        "gfx1100",
-                       "gfx1102",
-                       "gfx1031"};
```
Comment (atamazov) [inline test/handle_test.cpp:247]: Please roll back all gfx1031 related changes. Unofficial support for gfx1031 (Navi 22, Radeon RX6700) has been introduced in #1531

**Review by atamazov — CHANGES_REQUESTED** [review]
Please add "[CK]" to the title of this PR. It adds support for CK solvers only.

File: test/handle_test.cpp (hunk 1 above)
Comment (junliume) [reply test/handle_test.cpp:247]: @atamazov we are not removing it, just reordered the list :)

Comment (atamazov) [reply test/handle_test.cpp:247]: @junliume Oops 🤕

**Comment by atamazov** [comment]
@junliume If we want to add support for gfx1101 in tests, then pls look there:
- https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1925/files#r1089499601
- https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1925/files#r1089513155

**Review by atamazov — APPROVED** [review]
LGTM! It seems that it adds support for gfx1101 wherever it is necessary (except tests... for now) so maybe "[CK]" to the title is actually redundant ((

File: src/include/miopen/solver/ck_utility_common.hpp [hunk 2]
```diff
@@ -53,10 +53,11 @@ static inline bool is_ck_supported_hardware(const Handle& handle)
...
            StartsWith(handle.GetDeviceName(), "gfx908") ||
            StartsWith(handle.GetDeviceName(), "gfx90a") ||
-           StartsWith(handle.GetDeviceName(), "gfx1100") ||
-           StartsWith(handle.GetDeviceName(), "gfx1102") ||
+           StartsWith(handle.GetDeviceName(), "gfx1030") ||
            StartsWith(handle.GetDeviceName(), "gfx1031") ||
-           StartsWith(handle.GetDeviceName(), "gfx1030");
+           StartsWith(handle.GetDeviceName(), "gfx1100") ||
+           StartsWith(handle.GetDeviceName(), "gfx1101") ||
+           StartsWith(handle.GetDeviceName(), "gfx1102");
```
Comment (JehandadKhan) [inline src/include/miopen/solver/ck_utility_common.hpp:60]: Perhaps
``` C++
StartsWith(handle.GetDevicename(), "gfx11");
```

**Pull request merged**